internal/grpc/services/publicshareprovider/publicshareprovider.go

```diff
@@ -148,8 +149,17 @@ func (s *service) GetPublicShareByToken(ctx context.Context, req *link.GetPublic
 	log := appctx.GetLogger(ctx)
 	log.Debug().Msg("getting public share by token")
 
+	sig := req.GetAuthentication().GetSignature()
+	auth := publicshare.Authentication{
```

Comment:
Can you just pass `req.GetAuthentication()` to the package? This looks like we're unnecessarily complicating things